ios/FluentUI/Controls/ButtonVnext.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+public class ButtonVnextState: ObservableObject {
+	@Published var title: String = ""
+	@Published var isDisabled: Bool = false
+}
+
+public struct ButtonVnextViewButtonStyle: ButtonStyle {
+	var targetButton: ButtonVnextView
+
+	public init(targetButton: ButtonVnextView) {
+		self.targetButton = targetButton
+	}
+
+	public func makeBody(configuration: Self.Configuration) -> some View {
+		configuration.label
+			.foregroundColor(Color(targetButton.state.isDisabled ?
+									targetButton.tokens.disabledTitleColor :
+									(configuration.isPressed ?
+										targetButton.tokens.highlightedTitleColor :
+										targetButton.tokens.titleColor)))
+			.background(targetButton.tokens.hasBorders ?
+				AnyView(RoundedRectangle(cornerRadius: targetButton.tokens.cornerRadius)
+					.strokeBorder(lineWidth: targetButton.tokens.borderWidth, antialiased: false)
+					.foregroundColor(Color(targetButton.state.isDisabled ?
+											targetButton.tokens.disabledBorderColor :
+											(configuration.isPressed ?
+												targetButton.tokens.highlightedBorderColor :
+												targetButton.tokens.borderColor)))) :
+				AnyView(RoundedRectangle(cornerRadius: targetButton.tokens.cornerRadius)
+					.fill(Color(targetButton.state.isDisabled ?
+									targetButton.tokens.disabledBackgroundColor :
+									(configuration.isPressed ?
+										targetButton.tokens.highlightedBackgroundColor :
+										targetButton.tokens.backgroundColor))))
+				)
+	}
+}
+
+public struct ButtonVnextView: View {
+	var action: () -> Void
+	@ObservedObject var tokens: ButtonVnextAppearanceProxy
+	@ObservedObject var state: ButtonVnextState
+
+	public init(action: @escaping () -> Void, style: ButtonVnextStyle) {
+		self.action = action
```

Comment:
It feels weird that we have to wrap things that clients should be able to just send to the button. What if we just exposed a buttonStyle and not a buttonView itself?

Reply:
In this case we need to expose the Button view because we're changing its behavior based on the state (disabled/enabled), which is something that the style can't do.
But I still kept the View as slim as possible and moved the visual aspects to the ButtonStyle implementation.